htdp-test/2htdp/tests/abstraction-use.rkt

```diff
@@ -122,5 +122,16 @@
 
 (define (undress a-doll)
   (match a-doll
-    [(struct doll (inside)) (undress inside)]
+    [(doll inside) (undress inside)]
```

Comment:
Should `(struct posn (x y))` work?

```
(match (make-posn 1 2)
  [(struct posn (x y)) (+ x y)])
```

Note that currently, this code fails to expand.

Reply:
Interesting. I had checked in `#lang htdp/asl` and there it worked -- but that's because ASL re-exports the original `match`. 

We're fine then.

Reply:
Sorry, I'm a bit confused. What do you mean when you say "we're fine then"? Do you still want `match` from `2htdp/abstraction` to support patterns like `(struct doll (inside))`?

Note that

```
#lang htdp/asl

(match (make-posn 1 2)
  [(struct posn (x y)) (+ x y)])
```

works fine. That's because there's an implementation of [`match` for ASL](https://github.com/racket/htdp/blob/master/htdp-lib/lang/private/teach.rkt#L2864) that is hard coded to [recognize `posn`](https://github.com/racket/htdp/blob/master/htdp-lib/lang/private/teach.rkt#L2963).

On the other hand, by adding `(require 2htdp/abstraction)`

```
#lang htdp/asl

(require 2htdp/abstraction)

(match (make-posn 1 2)
  [(struct posn (x y)) (+ x y)])
```

the code now fails, because this `match` (which is the `match` that we are talking about) is re-exported from `racket/match`, so it doesn't recognize `posn` specially.

Reply:
Yes, I had checked these options, and I was trying to say what you have is good. 

This library doesn't find much usage from what I understand so we can actually enforce the grammar and use it to synthesize patterns.

Reply:
Got it, thanks!